core/dbt/include/global_project/macros/etc/statement.sql

```diff
@@ -1,13 +1,27 @@
-{% macro statement(name=None, fetch_result=False, auto_begin=True) -%}
+{#--
+TODO:  Only certian adapters need to support python models so this should probably be
+moved to an override I.E. `spark__statement`  but for some reason it wasn't picking up that change
+when implemented in dbt-spark/dbt/include/spark/macros/adapters.sql
+--#}
+
+{%- macro statement(name=None, fetch_result=False, auto_begin=True, language='sql') -%}
   {%- if execute: -%}
-    {%- set sql = caller() -%}
+    {%- set compiled_code = caller() -%}
 
     {%- if name == 'main' -%}
-      {{ log('Writing runtime SQL for node "{}"'.format(model['unique_id'])) }}
-      {{ write(sql) }}
+      {{ log('Writing runtime {} for node "{}"'.format(language, model['unique_id'])) }}
+      {{ write(compiled_code) }}
+    {%- endif -%}
+    {%- if language == 'sql'-%}
+      {%- set res, table = adapter.execute(compiled_code, auto_begin=auto_begin, fetch=fetch_result) -%}
+    {%- elif language == 'python' -%}
+      {%- set res = adapter.submit_python_job(model, compiled_code) -%}
+      {#-- TODO: What should table be for python models? --#}
```

Comment:
Question: Doesn't a Python job also generate a table on the data warehouse?

Reply:
Yes! This `table` refers to metadata returned when the query completes, which dbt uses to create [its "result" object](https://docs.getdbt.com/reference/artifacts/run-results-json). Depending on how we submit and return the query, that table can include information like status code and number of rows created/updated. It's orthogonal to the actual table (relational object) produced by the query in the database.

Reply:
@chamini2 I tool a swing at the 3 adapters we added python support for and realized that we don't have much info to give for now other than status. Snowflake will only return row affected as 1 for stored procedure, for databricks and dataproc on gcp there seems to be no such info right now.  We will probably circle back later on to see if things changes